packages/bundle-source/src/index.js

```diff
@@ -155,30 +159,124 @@ async function transformSource(
 
   // Now generate the sources with the new positions.
   return (babelGenerate.default || babelGenerate)(ast, {
+    sourceFileName: sourceMapUrl,
+    sourceMaps: true,
     retainLines: true,
     compact: true,
   });
 }
 
-async function bundleZipBase64(startFilename, dev, powers = {}) {
-  const entry = url.pathToFileURL(path.resolve(startFilename));
-  const { bytes, sha512 } = await makeAndHashArchive(
-    { ...readPowers, ...powers },
-    entry,
-    {
-      dev,
-      moduleTransforms: {
-        async mjs(sourceBytes) {
-          const source = textDecoder.decode(sourceBytes);
-          const { code: object } = await transformSource(source, {
-            sourceType: 'module',
-          });
-          const objectBytes = textEncoder.encode(object);
-          return { bytes: objectBytes, parser: 'mjs' };
-        },
+async function bundleZipBase64(startFilename, options = {}, powers = {}) {
+  const { dev = false, cache = true } = options;
+  const {
+    computeSha512,
+    pathResolve = path.resolve,
+    userInfo = os.userInfo,
+    env = process.env,
+    platform = process.platform,
+  } = powers;
+
+  const entry = url.pathToFileURL(pathResolve(startFilename));
+
+  const sourceMapJobs = new Set();
+  let writeSourceMap = Function.prototype;
+  if (cache) {
+    const { homedir: home } = userInfo();
```

Comment:
`bundleSource` is scorched earth for OCaps: it is already reaching for ambient authority and I did not attempt to improve on this as part of this change. My intention is to factor out `@endo/bundle-source/node-powers.js` and `@endo/bundle-source/powerless.js`  (or some such one-off naming convention for similar scorched-earth packages) in some future refactor.